http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala

```diff
@@ -479,53 +504,51 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
         if (creatorOnly.isDefined) {
           F.pure(runtimes)
         } else {
-          val (runtimesUserIsCreator, runtimesUserIsNotCreator) =
-            runtimes.partition(_.auditInfo.creator == userInfo.userEmail)
 
           // Here, we optimize the SAM lookups based on whether we will need to use the google project fallback
           // IF (workspaceId) EXISTS we need WSM resource ID sam lookup + workspaceId Fallback
           // ELSE we need googleProject fallback
-          val (runtimesUserIsNotCreatorWithWorkspaceId, runtimesUserIsNotCreatorWithoutWorkspaceId) =
-            runtimesUserIsNotCreator
+          val (runtimesUserWithWorkspaceId, runtimesUserWithoutWorkspaceId) =
+            runtimes
               .partition(_.workspaceId.isDefined)
 
           // Here, we check if backleo has updated the runtime sam id with the wsm resource's UUID via type conversion
           // If it has, we can then use the samResourceId of the runtime (which is the same as the wsm resource id) for permission lookup
 
-          // -----------   Filter runtimes that's in runtimesUserIsNotCreatorWithWorkspaceId   -----------
-          val runtimesUserIsNotCreatorWithWorkspaceSamIds = NonEmptyList
+          // -----------   Filter runtimes that's in runtimesUserWithWorkspaceId   -----------
+          val runtimesUserWithWorkspaceSamIds = NonEmptyList
             .fromList(
-              runtimesUserIsNotCreatorWithWorkspaceId.flatMap(runtime =>
+              runtimesUserWithWorkspaceId.flatMap(runtime =>
                 runtime.workspaceId match {
                   case Some(id) => List((runtime, WorkspaceResourceSamResourceId(id)))
                   case None     => List.empty
                 }
               )
             )
           for {
-            // If user is not creator for the runtime, then we check if they're workspace owner
-            runtimesUserIsNotCreatorWithWorkspaceIdAndSamResourceId <- runtimesUserIsNotCreatorWithWorkspaceSamIds
+            // We check if the user has access to the workspace
+            runtimesUserWithWorkspaceIdAndSamResourceId <- runtimesUserWithWorkspaceSamIds
               .traverse(workspaces =>
-                authProvider.filterWorkspaceOwner(
+                authProvider.filterWorkspaceReader(
```

Comment:
OK I think I see -- `filterWorkspaceReader` is just returning the resources from Sam (it's not doing any filtering). The filtering is occurring in the next call to `mapFilter`.

Seems like it was this way before, but this code feels quite confusing to me..

Reply:
Agreed, this function (and the list runtime one) is trying to do too much and could benefit from a larger refactoring